File: firebase/functions/src/model/BotDuel.ts
import { DocumentReference, Timestamp } from "firebase-admin/firestore";

export type BotDuel = {
    startTime: Timestamp,
    player: DocumentReference,
    playerBoard: string,
    endTime: Timestamp | null,
    given: string,
    botEndTime: Timestamp,
    botRanking: number,
    sudoku: DocumentReference,
}

File: app/ios/Sudoku Battles/Data/Model/Firebase/BotDuel.swift
import Foundation
import FirebaseFirestore

struct BotDuel: Codable {
    let startTime: Timestamp
    let player: DocumentReference
    let playerBoard: String
    let endTime: Timestamp?
    let given: String
    let botEndTime: Timestamp
    let sudoku: DocumentReference
    let botRanking: Int

    var difficulty: Difficulty { Difficulty(rawValue: sudoku.parent.collectionID) ?? .easy }
}
